### PR TITLE
Return structured errors from GetScanReport handler

### LINKED ISSUE #18: Provide error responses that are more than a text message and 500 status code
As specified in the Scanner Adapter API, each error should have a structure which can be parsed and handled by clients.

### PR DESCRIPTION
**Pull request by danielpacak**
Resolves #18 It was partially done for GetMetadata and AcceptScanRequest HTTP handlers. In this PR I did it for GetScanReport handler. Some changes where required to bubble up the error message from the MicroScanner stderr to the HTTP layer.

### CODE REVIEW

File: pkg/store/redis/store.go [hunk 1]
```diff
@@ -95,6 +95,21 @@ func (rs *redisStore) UpdateStatus(scanID string, currentStatus, newStatus job.S
 	return rs.SaveScanJob(scanJob)
 }
 
+func (rs *redisStore) UpdateAsFailed(scanJobID, message string) error {
```
Comment (lizrice) [inline pkg/store/redis/store.go]: Maybe better to do this with `UpdateStatus()` and a `newStatus` of `Failed`? If the worry is that you might not always have `currentStatus` correct, you could allow `currentStatus` to be anything provided that `newStatus` is `Failed`, wdyt?

File: pkg/http/api/v1/handler.go [hunk 2]
```diff
@@ -162,33 +163,48 @@ func (h *requestHandler) GetScanReport(res http.ResponseWriter, req *http.Reques
...
+		reqLog.WithField("scan_job_status", scanJob.Status).Debug("Scan job has not finished yet")
 		res.Header().Set(headerRefreshAfter, "15")
 		res.WriteHeader(http.StatusFound)
 		return
 	}
 
 	if scanJob.Status == job.Failed {
-		log.Errorf("Scan job failed: %v", scanJobID)
-		h.SendInternalServerError(res)
+		reqLog.Error("Scan job failed")
```
[the remark this answers is not in the document]
Comment (danielpacak) [reply pkg/http/api/v1/handler.go]: Very good point. Changed to:

```go
reqLog.WithField(log.ErrorKey, scanJob.Error).Error("Scan job failed")
```

File: pkg/store/redis/store.go (hunk 1 above)
Comment (danielpacak) [reply pkg/store/redis/store.go]: I like having just a single method for updating status as you pointed out. I'm just worried how to make the 4th arg optional. It makes sense to pass an error `message` only if the `newStatus` is `Failed`.

```go
UpdateStatus(scanJobID string, currentStatus, newStatus job.Status, message ...string)
```

If variadic function is fine in such case I'm happy to refactor it.

Comment (danielpacak) [reply pkg/store/redis/store.go]: A scan job might fail being in Queued and Pending state. Therefore I got rid of the currentStatus arg to the UpdateStatus method.

**Comment by danielpacak** [comment]
> I'd rather stick with `scan()` than `scanE()` but not too worried about it. We can come back to the recover() thing later :-)

Yeah. This thing is intriguing.

**Pull request merged**